Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/execution/CqlEngine.java

```diff
@@ -430,9 +432,23 @@ public Object visitUpper(Upper elm, State state) {
 
     @Override
     public Object visitUnion(Union elm, State state) {
-        Object left = visitExpression(elm.getOperand().get(0), state);
-        Object right = visitExpression(elm.getOperand().get(1), state);
-        return UnionEvaluator.union(left, right, state);
+        var left = elm.getOperand().get(0);
+        var right = elm.getOperand().get(1);
+        Object leftResult = visitExpression(left, state);
+        Object rightResult = visitExpression(right, state);
+
+
+        // This will attempt to use the declared result types from the ELM to determine which type of Union
+        // to perform. If the types are not declared, it will fall back to checking the values of the result.
+        if (left.getResultType() instanceof ListType || right.getResultType() instanceof ListType || elm.getResultType() instanceof ListType) {
+            return UnionEvaluator.unionIterable((Iterable<?>)leftResult, (Iterable<?>)rightResult, state);
+        }
+        else if (left.getResultType() instanceof IntervalType || right.getResultType() instanceof IntervalType || elm.getResultType() instanceof ListType) {
```

Comment:
Shouldn't this last `elm.getResultType() instanceof ListType` be IntervalType as well?

Reply:
Yep. You are right.